traitsui/testing/tester/_ui_tester_registry/qt4/_traitsui/table_editor.py

```diff
@@ -0,0 +1,117 @@
+    Selected,
+    SelectedIndices,
+)
+from traitsui.testing.tester._ui_tester_registry._common_ui_targets import (
+    BaseSourceWithLocation
+)
+from traitsui.testing.tester._ui_tester_registry.qt4 import (
+    _interaction_helpers
+)
+
+
+class _SimpleEditorWithCell(BaseSourceWithLocation):
+    source_class = SimpleEditor
+    locator_class = Cell
+    handlers = [
+        (MouseClick, lambda wrapper, _: wrapper._target._mouse_click(
+            delay=wrapper.delay)),
+        (KeyClick, lambda wrapper, interaction: wrapper._target._key_click(
+            key=interaction.key,
+            delay=wrapper.delay,)),
+        (
+            KeySequence,
+            lambda wrapper, interaction: wrapper._target._key_sequence(
+                sequence=interaction.sequence,
+                delay=wrapper.delay,
+            )
+        ),
+        (
+            DisplayedText,
+            lambda wrapper, _: wrapper._target._get_displayed_text()
+        ),
+        (MouseDClick, lambda wrapper, _: wrapper._target._mouse_dclick(
+            delay=wrapper.delay,)),
+    ]
+
+    def _get_model_view_index(self):
+        table_view = self.source.table_view
+        return dict(
+            model=table_view.model(),
+            view=table_view,
+            index=table_view.model().index(
+                self.location.row, self.location.column
+            ),
+        )
+
+    def _mouse_click(self, delay=0):
+        _interaction_helpers.mouse_click_item_view(
+            **self._get_model_view_index(),
+            delay=delay,
+        )
+
+    def _mouse_dclick(self, delay=0):
+        _interaction_helpers.mouse_dclick_item_view(
+            **self._get_model_view_index(),
+            delay=delay,
+        )
+
+    def _key_sequence(self, sequence, delay=0):
+        _interaction_helpers.key_sequence_item_view(
+            **self._get_model_view_index(),
+            sequence=sequence,
+            delay=delay,
+        )
+
+    def _key_click(self, key, delay=0):
+        _interaction_helpers.key_click_item_view(
+            **self._get_model_view_index(),
+            key=key,
+            delay=delay,
+        )
+
+    def _get_displayed_text(self):
+        return _interaction_helpers.get_display_text_item_view(
+            **self._get_model_view_index(),
+        )
+
+
+def register(registry):
+    """ Register interactions for the given registry.
+
+    If there are any conflicts, an error will occur.
+
+    Parameters
+    ----------
+    registry : TargetRegistry
+        The registry being registered to.
+    """
+    _SimpleEditorWithCell.register(registry)
+    registry.register_interaction(
+        target_class=SimpleEditor,
+        interaction_class=Selected,
+        handler=lambda wrapper, _: wrapper._target.selected
+    )
+    registry.register_interaction(
+        target_class=SimpleEditor,
+        interaction_class=SelectedIndices,
+        handler=lambda wrapper, _: wrapper._target.selected_indices
+    )
```

Comment:
I've just added a `SelectedIndices` query class.
For the current implementation, for `TableEditor`, `Selected` and `SelectedIndices` effectively just serve as proxies to the `selected` and `selected_indices` traits on the table editor (see code above.)
In theory it may be better for these to dive down to the Qt level (?)

Reply:
This is one of the weird things with UITester support though IMO.  We want to test traitsui editors are working correctly.
So having tests be down at qt level is really what we want. Ie I do something in traitsUI and we verify that does what we expect down in qt land.
However, say I were implementing the handler for the TableEditor to do a `SelectedIndices` query. I would end up basically rewriting:
https://github.com/enthought/traitsui/blob/347f0b5df25b4432b1163c34ebfd8a99c5a26ae0/traitsui/qt4/table_editor.py#L626-L643
It creates sort of a weird chasing your own tail scenario 🤔 

For downstream users they just want to see that when they run _their_ code, ____ is selected.  For which this (just using traits on the editor) is perfectly fine.

Reply:
`UITester` strikes me as being useful for two things:
- testing that TraitsUI editors are integrating correctly with the toolkit
- as a framework for integration tests for TraitsUI apps

For unit tests of applications you can usually get away with tests at the traits level (does changing this trait in a UI change the corresponding thing in the model? or vice-versa) where you don't really care/can trust that the UI then does the right thing.